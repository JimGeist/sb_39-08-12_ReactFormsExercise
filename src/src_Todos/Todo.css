.Todo-BtnDelete {
    padding: 1px 10px 1px 10px;
    margin: 2px 0 0 10px;
}