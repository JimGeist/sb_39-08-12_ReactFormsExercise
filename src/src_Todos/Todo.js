import React from "react"

import "./Todo.css";

function Todo({ id, todo, deleteFx }) {

    const handleDelete = () => deleteFx(id)

    return (
        <li className="Todo-li" data-testid="^Todo Item^" >
            {todo}
            <button className="Todo-BtnDelete" onClick={handleDelete}>X</button>
        </li>
    )

}

export default Todo;