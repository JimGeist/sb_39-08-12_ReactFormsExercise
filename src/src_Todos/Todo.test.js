import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

// Smoke Test -- does Todo render?
it("renders the Todo component without crashing", () => {
    render(<Todo />);
});

// Todo snapshot tests
it("Todo component without props matches snapshot", () => {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

it("Todo component with test todo item matches snapshot", () => {
    const { asFragment } = render(<Todo id={1627181188638} todo="This is a test todo message" />);
    expect(asFragment()).toMatchSnapshot();
});
