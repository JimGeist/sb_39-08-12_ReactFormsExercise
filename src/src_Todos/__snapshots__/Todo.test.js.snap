// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component with test todo item matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="Todo-li"
    data-testid="^Todo Item^"
  >
    This is a test todo message
    <button
      class="Todo-BtnDelete"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;

exports[`Todo component without props matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="Todo-li"
    data-testid="^Todo Item^"
  >
    <button
      class="Todo-BtnDelete"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
